crypto/src/e2e_identity/mod.rs

```diff
@@ -83,6 +100,184 @@ impl MlsCentral {
+        self.e2ei_pki_env.replace(pki_env);
+
+        Ok(())
+    }
+
+    /// Registers a Root Trust Anchor CA for the use in E2EI processing.
+    ///
+    /// Please note that without a Root Trust Anchor, all validations *will* fail;
+    /// So this is the first step to perform after initializing your E2EI client
+    ///
+    /// # Parameters
+    /// * `trust_anchor_pem` - PEM certificate to anchor as a Trust Root
+    pub async fn e2ei_register_acme_ca(&mut self, trust_anchor_pem: String) -> CryptoResult<()> {
+        // Bail if we have an environment already setup
+        if self.e2ei_pki_env.is_some() {
+            return Err(CryptoError::ConsumerError);
+        };
+
+        let pki_env = PkiEnvironment::init(PkiEnvironmentParams {
+            intermediates: Default::default(),
+            trust_roots: Default::default(),
+            crls: Default::default(),
+            time_of_interest: Default::default(),
+        })
+        .map_err(|e| CryptoError::E2eiError(e.into()))?;
+
+        // Parse/decode PEM cert
+        let root_cert =
+            PkiEnvironment::decode_pem_cert(trust_anchor_pem).map_err(|e| CryptoError::E2eiError(e.into()))?;
+
+        // Validate it (expiration & signature only)
+        pki_env
+            .validate_trust_anchor_cert(&root_cert)
+            .map_err(|e| CryptoError::E2eiError(e.into()))?;
+
+        // Save DER repr in keystore
+        let cert_der = PkiEnvironment::encode_cert_to_der(&root_cert).map_err(|e| CryptoError::E2eiError(e.into()))?;
+        let acme_ca = E2eiAcmeCA { content: cert_der };
+        let mut conn = self.mls_backend.key_store().borrow_conn().await?;
+        acme_ca.replace(&mut conn).await?;
+        drop(conn);
+
+        // To do that, tear down and recreate the pki env
+        self.init_pki_env().await?;
+
+        Ok(())
+    }
+
+    /// Registers an Intermediate CA for the use in E2EI processing.
+    ///
+    /// Please note that a Root Trust Anchor CA is needed to validate Intermediate CAs;
+    /// You **need** to have a Root CA registered before calling this
+    ///
+    /// # Parameters
+    /// * `cert_pem` - PEM certificate to register as an Intermediate CA
+    pub async fn e2ei_register_intermediate_ca(&mut self, cert_pem: String) -> CryptoResult<()> {
+        if self.e2ei_pki_env.is_none() {
+            return Err(CryptoError::ConsumerError);
+        };
+
+        // Parse/decode PEM cert
+        let inter_ca = PkiEnvironment::decode_pem_cert(cert_pem).map_err(|e| CryptoError::E2eiError(e.into()))?;
+        let (ski, aki) =
+            PkiEnvironment::extract_ski_aki_from_cert(&inter_ca).map_err(|e| CryptoError::E2eiError(e.into()))?;
+
+        let ski_aki_pair = format!("{ski}:{}", aki.unwrap_or_default());
+
+        // Validate it
+        let Some(pki_env) = self.e2ei_pki_env.as_ref() else {
+            unreachable!("Pki Env mysteriously disappeared despite being checked beforehand");
+        };
```

Comment:
Move this to the beginning of the method instead of checking it first then panicking when it's not found

Reply:
Good catch